File: 1961.py
pulos, n = input().split(" ")
canos = input().split(" ")
l1 = int(canos[0])
vitoria = 0
for i in range(1, int(n)):
    l2 = int(canos[i])
    if l2 > l1 and l2 - l1 > int(pulos):
        vitoria = 1
        break
    if l1 > l2 and l1 - l2 > int(pulos):
        vitoria = 1
        break
    l1 = int(canos[i])
if vitoria == 0:
    print('YOU WIN')
else:
    print('GAME OVER')

File: 1961.cpp
#include <iostream>
#include <cmath>
using namespace std;

int main(){
    int altura_pulo, numero_canos;
    cin >> altura_pulo >> numero_canos;
    int canos[numero_canos];
    cin >> canos[0];
    for(int i = 1; i < numero_canos; i++){
        cin >> canos[i];
        if(max(canos[i-1], canos[i]) - min(canos[i-1], canos[i]) > altura_pulo){
            cout << "GAME OVER\n";
            return 0;
        }
    }
    cout << "YOU WIN\n";
    return 0;
}
